testsuite/MDAnalysisTests/plugins/open_files.py

```diff
@@ -0,0 +1,79 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4 fileencoding=utf-8
+#
+# MDAnalysis --- http://www.MDAnalysis.org
+# Copyright (c) 2006-2015 Naveen Michaud-Agrawal, Elizabeth J. Denning,
+# Oliver Beckstein and contributors (see AUTHORS for the full list)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+
+from __future__ import print_function
+
+import collections
+import os
+
+import psutil
+
+from nose.plugins.base import Plugin
+from nose.pyversion import exc_to_unicode, force_unicode
+from nose.util import ln
+
+
+class ReportOpenFiles(Plugin):
+    enabled = True
+    env_opt = 'NOSE_NO_MDA_FILES'
+    name = 'open-files'
+
+    def options(self, parser, env):
+        """Registers the commandline option, defaulting to enabled.
+        """
+        parser.add_option(
+            "--no-{}".format(self.name), action="store_false",
+            default=not env.get(self.env_opt), dest="do_mda_open_files",
+            help="Don't display open file handlers. [{}]".format(self.env_opt))
+
+    def configure(self, options, conf):
+        super(ReportOpenFiles, self).configure(options, conf)
+        self.config = conf # This will let other tests know about config settings.
+        try:
+            self.enabled = options.do_mda_open_files
+        except AttributeError:
+            self.enabled = False
+
+    def formatError(self, test, err):
+        open_files = [f.path for f in self._gather_open_files()]
+
+        if not open_files:
+            return err
+
+        ec, ev, tb = err
+        ev = exc_to_unicode(ev)
+        handle_couter = collections.Counter(open_files)
+        new_ev = u'\n'.join(
+            [ev]
+            + [ln(u'>>> There {} open file handlers for {} files: <<<'
```

Comment:
You can even iterate over the set and count the different reader types to check if they match up with the numbers of open filetypes.